hazelcast/src/main/java/com/hazelcast/internal/metrics/metricsets/StatisticsAwareMetricsSet.java

```diff
@@ -70,13 +74,17 @@ private Task(MetricsRegistry metricsRegistry) {
 
         @Override
         public void run() {
-            registerAliveStats();
-            purgeDeadStats();
-
-            Set<LocalInstanceStats> tmp = previousStats;
-            previousStats = currentStats;
-            currentStats = tmp;
-            currentStats.clear();
+            try {
+                registerAliveStats();
+                purgeDeadStats();
+
+                Set<LocalInstanceStats> tmp = previousStats;
+                previousStats = currentStats;
+                currentStats = tmp;
+                currentStats.clear();
+            } catch (Exception e) {
+                logger.finest("Error occurred while scanning for statistics aware metrics", e);
```

Comment:
Would it not be better to fix the logging in the Diagnostics thread? So that every plugin can benefit from this exception protection.

Reply:
Which thread is that? On the scheduled executor? Might be. But I'm not sure if I want to apply the same decision - rerun the task again for all current and future tasks submitted to the executor. What if one task wants to stop working? What if a different one needs to cleanup its state?

Reply:
Let's agree on the smallest scope for now plz.